app/controllers/newbookings_controller.rb

```diff
@@ -6,29 +6,32 @@ def show
         @newbooking = Newbooking.find(params[:id])
       end
     def new
-        @newbooking = Newbooking.new
+        @customer = Customer.find(params[:customer_id])
```

Comment:
this is a good start, but there is some easy improvements in here.

DRY (don't repeat yourself) is a useful thing to remember. If you're writing the same few lines of code more than once, extract the code to a method and call that method instead. The reasons we would want to do that include:
* less code to look at and understand
* less code we need to maintain and test
* if we change things in the future, we only have to change them in one place

In this class, we are finding the customer and pet in several methods in the same way. This means that we could benefit from a new method:

```ruby
def find_customer_and_pet
  @customer = Customer.find...
  @pet = Pet.find...
end
```
Then in the actions, we could just call `find_customer_and_pet` instead of the several lines we're calling now.

Once we have that method separated out, it let's us do something else so we can DRY even more. If we have that method, we can add to the top of the file something like `before_action :find_customer_and_pet, only: [:new, :create]`. What that does is runs that new method before your action method runs... so you don't even need to call it!

Reply:
I see, that makes sense!